geomesa-raster/src/main/scala/org/locationtech/geomesa/raster/util/RasterUtils.scala

```diff
@@ -0,0 +1,291 @@
+    val IS_TEST_INGEST      = "geomesa.tools.ingestraster.is-test-ingest"
+  }
+
+  def imageSerialize(image: RenderedImage): Array[Byte] = {
+    val buffer: ByteArrayOutputStream = new ByteArrayOutputStream
+    val out: ObjectOutputStream = new ObjectOutputStream(buffer)
+    val serializableImage = new SerializableRenderedImage(image, true)
+    try {
+      out.writeObject(serializableImage)
+    } finally {
+      out.close
+    }
+    buffer.toByteArray
+  }
+
+  def imageDeserialize(imageBytes: Array[Byte]): RenderedImage = {
+    val in: ObjectInputStream = new ObjectInputStream(new ByteArrayInputStream(imageBytes))
+    var read: RenderedImage = null
+    try {
+      read = in.readObject.asInstanceOf[RenderedImage]
+    } finally {
+      in.close
+    }
+    read
+  }
+
+  def allocateBufferedImage(width: Int, height: Int, chunk: RenderedImage): BufferedImage = {
+    val properties = new JHashtable[String, Object]
+    if (chunk.getPropertyNames != null) {
+      chunk.getPropertyNames.foreach(name => properties.put(name, chunk.getProperty(name)))
+    }
+    val colorModel = chunk.getColorModel
+    val alphaPremultiplied = colorModel.isAlphaPremultiplied
+    val sampleModel = chunk.getSampleModel.createCompatibleSampleModel(width, height)
+    val emptyRaster = JRaster.createWritableRaster(sampleModel, null)
+    new BufferedImage(colorModel, emptyRaster, alphaPremultiplied, properties)
+  }
+
+  def allocateBufferedImage(width: Int, height: Int, buf: BufferedImage): BufferedImage = {
+    val colorModel = buf.getColorModel
+    val alphaPremultiplied = colorModel.isAlphaPremultiplied
+    val emptyRaster = colorModel.createCompatibleWritableRaster(width, height)
+    new BufferedImage(colorModel, emptyRaster, alphaPremultiplied, null)
+  }
+
+  def renderedImageToBufferedImage(r: RenderedImage): BufferedImage = {
+    val properties = new JHashtable[String, Object]
+    if (r.getPropertyNames != null) {
+      r.getPropertyNames.foreach(name => properties.put(name, r.getProperty(name)))
+    }
+    val colorModel = r.getColorModel
+    val alphaPremultiplied = colorModel.isAlphaPremultiplied
+    val sampleModel = r.getSampleModel
+    new BufferedImage(colorModel, r.copyData(null), alphaPremultiplied, properties)
+  }
+
+  def writeToMosaic(mosaic: BufferedImage, raster: Raster, env: Envelope, resX: Double, resY: Double) = {
+    val croppedRaster = cropRaster(raster, env)
+    croppedRaster.foreach{ cropped =>
+        val rasterEnv = raster.referencedEnvelope.intersection(envelopeToReferencedEnvelope(env))
+        val originX = Math.floor((rasterEnv.getMinX - env.getMinimum(0)) / resX).toInt
+        val originY = Math.floor((env.getMaximum(1) - rasterEnv.getMaxY) / resY).toInt
+        mosaic.getRaster.setRect(originX, originY, cropped.getData)
+    }
+  }
+
+  def mosaicChunks(chunks: Iterator[Raster], queryWidth: Int, queryHeight: Int, queryEnv: Envelope): (BufferedImage, Int) = {
```

Comment:
[suggestion] Returning a case class with the image and the count as argument vals might be a tad more readable